toolkit/scripts/safeunmount.sh

```diff
@@ -2,23 +2,46 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
-# $1 path to recursively scan for mount points which must be cleaned up
-dir=${1}
-for dir in $(find ${dir} -type d | sort) ; do
-    if [[ -d $dir ]]; then
-        if  mountpoint -q ${dir} ; then
-            echo "WARNING: Removing mountpoint at $dir"
-            umount -l ${dir}
-            sleep 0.5
+# Usage: safeunmount.sh <dir1> <dir2> ...
+# e.g.
+#   `safeunmount.sh /mnt/resource /mnt/scratch` will attempt to unmount all mount points under /mnt/resource and /mnt/scratch in parallel
+#   `safeunmount.sh "/mnt/"*` will use bash expansion to unmount all mount points under /mnt/ in parallel
+
+function clean_dir {
+    dir=${1}
+    for dir in $(find "${dir}" -type d | sort) ; do
+        if [[ -d $dir ]]; then
+            if  mountpoint -q "${dir}" ; then
+                echo "WARNING: Removing mountpoint at $dir"
+                umount -l "${dir}"
+                sleep 0.5
+            fi
+            retries=10
+            while mountpoint -q "${dir}" ; do
+                echo "ERROR: Mountpoint still present at $dir, retrying unmount ${retries} times"
+                umount -l "${dir}"
+                retries=$(( "${retries}" - 1))
+                sleep 1
+                if [ ${retries} -eq 0 ] ; then return 1 ; fi
+            done
         fi
-        retries=10
-        while mountpoint -q ${dir} ; do
-            echo "ERROR: Mountpoint still present at $dir, retrying unmount ${retries} times"
-            umount -l ${dir}
-            retries=$(( ${retries} - 1))
-            sleep 1
-            if [ ${retries} -eq 0 ] ; then exit 1 ; fi
-        done
+    done || return 1
```

Comment:
Why do we have a `return 1` here? What errors, besides umount failure, can occur?

Reply:
The loop itself can error out if the input is garbage, and I've found error handling of background tasks unpredictable at the best of times. I tend to try and be explicit with return codes as much as possible. Maybe another approach would be to assume `return 1`, and only `return 0` if a final call to `mountpoint -q` says we are unmounted.

Reply:
Moved to https://github.com/microsoft/CBL-Mariner/pull/6262